corehq/ex-submodules/casexml/apps/stock/models.py

```diff
@@ -61,6 +61,14 @@ def __unicode__(self):
             case=self.case_id, product=self.product_id, section_id=self.section_id,
         )
 
+    def save(self, force_insert=False, force_update=False, using=None, update_fields=None):
+        # this is a bit hacky, but allows us to create StockReport and StockTransaction objects
+        # in a place that isn't exactly where we save them, which fits better with the current
+        # form processing workflows
+        if self.report_id is None and self.report and self.report.id is not None:
+            self.report_id = self.report.id
+        super(StockTransaction, self).save(force_insert=force_insert)
```

Comment:
This seems weird. `self.report_id` is the foreign key column so calling `self.report` will hit the DB if the model is saved. If not then you shouldn't need to set `self.report_id` separately, setting `self.report` is equivalent.